### DISCUSSION TITLE
entityChangeState Handler

### DISCUSSION THREAD

**Discussion by drakub** [YetiForceCompany/YetiForceCRM#15998] [General]
Hello team,

I have seen a handler (entitChangeState). I assume it is triggered when a picklist value changes on the form or a change event in any of the fields. If this assumption is right, how do I create such a handler? An example I can reference on??

**Comment by davide-alghi** [comment]
Hi @job-bob-draku,
no, that event is triggered when a record (an entity) change state: active, trashed, archived.

https://github.com/YetiForceCompany/YetiForceCRM/blob/ee251d7b2b5c4d8a018a807f05178de3751545c6/modules/Vtiger/actions/State.php#L44

https://github.com/YetiForceCompany/YetiForceCRM/blob/ee251d7b2b5c4d8a018a807f05178de3751545c6/modules/Vtiger/models/Record.php#L1798
